source/server/admin/pprof_handler.cc

```diff
@@ -0,0 +1,204 @@
+#include "source/server/admin/pprof_handler.h"
+
+#include <string>
+
+#include "source/common/common/assert.h"
+#include "source/server/admin/profile.pb.h"
+
+#include "absl/debugging/symbolize.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/str_format.h"
+#if defined(TCMALLOC)
+#include "tcmalloc/malloc_extension.h"
+#endif
+#include "zlib.h"
+
+namespace Envoy {
+namespace Server {
+
+namespace {
+
+#if defined(TCMALLOC)
+constexpr bool kHasTcMalloc = true;
+#else
+constexpr bool kHasTcMalloc = false;
+#endif
+
+constexpr size_t kChunkSize = 16 * 1024;
+constexpr int kGzipEncoding = 16;
+constexpr int kDefaultMemLevel = 8;
+
+class ZStream {
```

Comment:
I would recommend using https://github.com/envoyproxy/envoy/blob/main/source/extensions/compression/gzip/compressor/zlib_compressor_impl.h#L18 for compression if possible in order to avoid another direct dependency on zlib.